security/pkg/server/ca/server.go

```diff
@@ -238,9 +240,21 @@ func NewWithGRPC(grpc *grpc.Server, ca CertificateAuthority, ttl time.Duration,
 	authenticators := []authenticator{&authenticate.ClientCertAuthenticator{}}
 	serverCaLog.Info("added client certificate authenticator")
 
+	rc, err := rest.InClusterConfig()
+	if err != nil {
+		serverCaLog.Errorf("failed to get in cluster config: %v", err)
+		rc = &rest.Config{
+			Host: k8sAPIServerURL,
+			TLSClientConfig: rest.TLSClientConfig{
+				CAFile: caCertPath,
+			},
+			BearerTokenFile: jwtPath,
+		}
+	}
+
 	// Only add k8s jwt authenticator if SDS is enabled.
 	if sdsEnabled {
-		authenticator, err := authenticate.NewKubeJWTAuthenticator(k8sAPIServerURL, caCertPath, jwtPath,
+		authenticator, err := authenticate.NewKubeJWTAuthenticator(rc.Host+k8sAPIServerPath, rc.CAFile, rc.BearerTokenFile,
```

Comment:
Could you keep the original code path and while adding the new function call? This change might break existing customers when they upgrade their Istio deployment.
Please also add unit tests and integration test to cover the code change.